lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/storage/AzureStorageUtil.java

```diff
@@ -0,0 +1,150 @@
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.InvalidKeyException;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.EnumSet;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Azure storage utility to upload Driver and Evaluator jars to blob storage
+ * and generate SAS URIs.
+ */
+public class AzureStorageUtil {
+  private static final Logger LOG = Logger.getLogger(AzureStorageUtil.class.getName());
+
+  private static final int BLOB_SAS_TOKEN_VALIDITY_MINUTES = 30;
+  private static final int CONTAINER_SAS_TOKEN_VALIDITY_HOURS = 24;
+
+  private final ICloudBlobClientProvider cloudBlobClientProvider;
+
+  private final String azureStorageContainerName;
+
+  @Inject
+  AzureStorageUtil(
+      final ICloudBlobClientProvider cloudBlobClientProvider,
+      @Parameter(AzureStorageContainerName.class) final String azureStorageContainerName) {
+    this.cloudBlobClientProvider = cloudBlobClientProvider;
+    this.azureStorageContainerName = azureStorageContainerName;
+  }
+
+  public URI getJobSubmissionFolderUri(final String jobFolder) throws IOException {
+    final CloudBlobClient cloudBlobClient = this.cloudBlobClientProvider.getCloudBlobClient();
+    try {
+      final CloudBlobContainer container = cloudBlobClient.getContainerReference(this.azureStorageContainerName);
+      return container.getDirectoryReference(jobFolder).getUri();
+    } catch (URISyntaxException | StorageException e) {
+      throw new IOException("Failed to get the job submission folder URI", e);
+    }
+  }
+
+
+  public String createContainerSharedAccessSignature() throws IOException {
+    try {
+      CloudBlobClient cloudBlobClient = this.cloudBlobClientProvider.getCloudBlobClient();
+      CloudBlobContainer cloudBlobContainer = cloudBlobClient.getContainerReference(this.azureStorageContainerName);
+      cloudBlobContainer.createIfNotExists();
+
+      return cloudBlobContainer.generateSharedAccessSignature(getSharedAccessContainerPolicy(), null);
+
+    } catch (StorageException | URISyntaxException | InvalidKeyException e) {
+      throw new IOException("Failed to generate a shared access signature for storage container.", e);
+    }
+  }
+
+
+  /**
+   * Upload a file to the storage account.
+   *
+   * @param jobFolder the path to the destination folder within storage container.
+   * @param file the source file.
+   * @return the SAS URI to the uploaded file.
+   * @throws IOException
+   */
+  public URI uploadFile(final String jobFolder, final File file) throws IOException {
+
+    LOG.log(Level.INFO, "Uploading [{0}] to [{1}]", new Object[]{file, jobFolder});
+
+    try {
+      final CloudBlobClient cloudBlobClient = this.cloudBlobClientProvider.getCloudBlobClient();
+      final CloudBlobContainer container = cloudBlobClient.getContainerReference(this.azureStorageContainerName);
```

Comment:
Call "container.createIfNotExists();" should be consistent with the other functions instead of assuming an order or calling.

Reply:
sure